scripts/skills/perks/perk_rf_fresh_and_furious.nut

```diff
@@ -45,11 +48,31 @@ this.perk_rf_fresh_and_furious <- ::inherit("scripts/skills/skill", {
 
 	function isEnabled()
 	{
-		return this.getContainer().getActor().getFatigue() < 0.3 * this.getContainer().getActor().getFatigueMax();
+		return !this.m.RequiresRecover && this.getContainer().getActor().getFatigue() < this.m.FatigueThreshold * this.getContainer().getActor().getFatigueMax();
 	}
 
 	function onAfterUpdate( _properties )
 	{
+		// We do it like this instead of checking it in onTurnStart directly because some skills e.g. Encumbrance
```

Comment:
Unfortunately it's not just encumbrance i.e. we can't predict what other effects can be added by submods which modify the Fatigue value of the actor in `onTurnStart`. And we cannot predict those effects' skill order. Therefore, this is the safest bet.

Reply:
By that same logic we can't predict what other skills might also do this same work-around in order to apply some elaborate fatigue changes, e.g. when certain fatigue thresholds are met during start up.

Someone might implement a skill that recovers 5 Fatigue on turn start if you have at least X fatigue build up. And then they employ the same tech as you just did up there.

---

I much rather prefer lowering the skill order of encumbrance, rather than implementing these work around in every skill which checks any play resource on turn start.

Anyone who implements resource manipulating effects on turn start need to learn the easy/hard way that those skills need a lower skill order or other effects will not recognize those changes.